api/src/main/java/org/bahmni/module/bahmni/ie/apps/service/impl/BahmniFormPrivilegesServiceImpl.java

```diff
@@ -0,0 +1,151 @@
+//                for (int i = 0; i < oldPrivilegeList.size(); i++) {
+//                    resultList.add(bahmniFormPrivilegeDao.deleteFormPrivilege(oldPrivilegeList.get(i)));
+//                }
+//            }
+//            return resultList;
+//        } else {
+
+            Iterator privilegeItr = formPrivileges.iterator();
+            int formId = 0;
+            String formVersion = "";
+            log.info("Inside BahmniFormPrivilegesServiceImpl -->saveFormPrivileges " + privilegeItr);
+            while (privilegeItr.hasNext()) {
+                LinkedHashMap temp = (LinkedHashMap) privilegeItr.next();
+                List<Map.Entry> entrySetList = new ArrayList<Map.Entry>(temp.entrySet());
+                FormPrivilege tempPrivilege = new FormPrivilege();
+                for (Map.Entry tempEntrySet : entrySetList) {
+                    if (tempEntrySet.getKey().toString().equalsIgnoreCase("formId")) {
+                        tempPrivilege.setFormId((Integer) temp.get("formId"));
+                        formId = (Integer) temp.get("formId");
+                    } else if (tempEntrySet.getKey().toString().equalsIgnoreCase("editable")) {
+                        tempPrivilege.setEditable((Boolean) temp.get("editable"));
+                    } else if (tempEntrySet.getKey().toString().equalsIgnoreCase("privilegeName")) {
+                         String privilegeName = temp.get("privilegeName").toString();
+                         tempPrivilege.setPrivilegeName(privilegeName);
+                    } else if (tempEntrySet.getKey().toString().equalsIgnoreCase("viewable")) {
+                        tempPrivilege.setViewable((Boolean) temp.get("viewable"));
+                    }else if (tempEntrySet.getKey().toString().equalsIgnoreCase("formVersion")) {
+                        tempPrivilege.setFormVersion((String) temp.get("formVersion"));
+                        formVersion = tempPrivilege.getFormVersion();
+                    }
+                }
+                privilegesList.add(tempPrivilege);
+            }
+            if (privilegesList.size() == 1 && privilegesList.get(0).getPrivilegeName().equalsIgnoreCase("")) {
+                deleteAllThePrivilegesFromDB(formId , formVersion);
+            } else {
+                oldPrivilegeList = getAllPrivilegesForForm(formId , formVersion);
+                if ((oldPrivilegeList != null) && !(oldPrivilegeList.isEmpty())) {
+                    for (int i = 0; i < oldPrivilegeList.size(); i++) {
+                        bahmniFormPrivilegeDao.deleteFormPrivilege(oldPrivilegeList.get(i));
+                    }
+                }
+                    if (privilegesList != null) {
+                        for(int i=0;i<privilegesList.size();i++){
+                            resultList.add(saveFormPrivilege(privilegesList.get(i)));
+                        }
+                    }
+            }
+        return resultList;
+
+    }
+        @Transactional(readOnly = true)
```

Comment:
@buvaneswari-arun can you please check the code formatting here

Reply:
Done